<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CNC Tool</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_file_mode">
       <attribute name="title">
        <string>File Mode</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="0" rowspan="5">
           <widget class="QPlainTextEdit" name="pte_file_preview">
            <property name="font">
             <font>
              <family>Cascadia Mono Light</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="lineWrapMode">
             <enum>QPlainTextEdit::NoWrap</enum>
            </property>
            <property name="tabStopDistance">
             <double>40.000000000000000</double>
            </property>
            <property name="placeholderText">
             <string>File Preview / Manual Input</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1" colspan="2">
           <spacer name="vs_functions_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="3" column="1" colspan="2">
           <widget class="QPushButton" name="pb_save_file">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Save File</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLineEdit" name="le_file_path">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>File Path</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QToolButton" name="tb_reload_file">
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="toolTip">
             <string>Reload File</string>
            </property>
            <property name="text">
             <string>тно</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="pb_open_file">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Open File</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QGroupBox" name="gb_functions">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Functions</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="9" column="0" colspan="2">
              <widget class="Line" name="ln_functions_2">
               <property name="frameShadow">
                <enum>QFrame::Plain</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="sb_number_lines_start">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="prefix">
                <string>N</string>
               </property>
               <property name="maximum">
                <number>100000</number>
               </property>
               <property name="singleStep">
                <number>100</number>
               </property>
               <property name="value">
                <number>1000</number>
               </property>
              </widget>
             </item>
             <item row="7" column="0">
              <widget class="QLabel" name="lb_number_ids_step">
               <property name="text">
                <string>Step:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="lb_number_lines_step">
               <property name="text">
                <string>Step:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0" colspan="2">
              <widget class="Line" name="ln_functions_1">
               <property name="frameShadow">
                <enum>QFrame::Plain</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="lb_number_lines_start">
               <property name="text">
                <string>Start:</string>
               </property>
              </widget>
             </item>
             <item row="6" column="0">
              <widget class="QLabel" name="lb_number_ids_start">
               <property name="text">
                <string>Start:</string>
               </property>
              </widget>
             </item>
             <item row="8" column="0" colspan="2">
              <widget class="QPushButton" name="pb_number_ids">
               <property name="text">
                <string>Number IDS</string>
               </property>
              </widget>
             </item>
             <item row="6" column="1">
              <widget class="QSpinBox" name="sb_number_ids_start">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="prefix">
                <string>IDS=</string>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="singleStep">
                <number>10</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item row="13" column="0" colspan="2">
              <widget class="QPushButton" name="pb_convert_tabs_to_spaces">
               <property name="text">
                <string>Convert Tabs To Spaces</string>
               </property>
              </widget>
             </item>
             <item row="12" column="0">
              <widget class="QLabel" name="lb_tab_width">
               <property name="text">
                <string>Tab Width</string>
               </property>
              </widget>
             </item>
             <item row="12" column="1">
              <widget class="QSpinBox" name="sb_tab_width">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="maximum">
                <number>8</number>
               </property>
               <property name="singleStep">
                <number>2</number>
               </property>
               <property name="value">
                <number>4</number>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="sb_number_lines_step">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="singleStep">
                <number>10</number>
               </property>
               <property name="value">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="7" column="1">
              <widget class="QSpinBox" name="sb_number_ids_step">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>1</number>
               </property>
              </widget>
             </item>
             <item row="3" column="0" colspan="2">
              <widget class="QPushButton" name="pb_number_lines">
               <property name="text">
                <string>Number Lines</string>
               </property>
              </widget>
             </item>
             <item row="15" column="0" colspan="2">
              <widget class="QPushButton" name="pb_remove_trailing_whitespace">
               <property name="text">
                <string>Remove Trailing Whitespace</string>
               </property>
              </widget>
             </item>
             <item row="14" column="0" colspan="2">
              <widget class="Line" name="ln_functions_3">
               <property name="frameShadow">
                <enum>QFrame::Plain</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="2" column="1" colspan="2">
           <spacer name="vs_functions_1">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>10</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="4" column="1" colspan="2">
           <widget class="QPushButton" name="pb_save_file_as">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Save File As</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_folder_mode">
       <attribute name="title">
        <string>Folder Mode</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLineEdit" name="le_folder_path">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>Folder Path</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QToolButton" name="tb_reload_folder">
            <property name="font">
             <font>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="toolTip">
             <string>Reload Folder</string>
            </property>
            <property name="text">
             <string>тно</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="pb_open_folder">
            <property name="font">
             <font>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Open Folder</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QGroupBox" name="gb_functions_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Functions</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="9" column="0" colspan="2">
              <widget class="Line" name="ln_functions_4">
               <property name="frameShadow">
                <enum>QFrame::Plain</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="sb_number_lines_start_2">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="prefix">
                <string>N</string>
               </property>
               <property name="maximum">
                <number>100000</number>
               </property>
               <property name="singleStep">
                <number>100</number>
               </property>
               <property name="value">
                <number>1000</number>
               </property>
              </widget>
             </item>
             <item row="7" column="0">
              <widget class="QLabel" name="lb_number_ids_step_2">
               <property name="text">
                <string>Step:</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="lb_number_lines_step_2">
               <property name="text">
                <string>Step:</string>
               </property>
              </widget>
             </item>
             <item row="4" column="0" colspan="2">
              <widget class="Line" name="ln_functions_5">
               <property name="frameShadow">
                <enum>QFrame::Plain</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="lb_number_lines_start_2">
               <property name="text">
                <string>Start:</string>
               </property>
              </widget>
             </item>
             <item row="6" column="0">
              <widget class="QLabel" name="lb_number_ids_start_2">
               <property name="text">
                <string>Start:</string>
               </property>
              </widget>
             </item>
             <item row="8" column="0" colspan="2">
              <widget class="QPushButton" name="pb_number_ids_2">
               <property name="text">
                <string>Number IDS</string>
               </property>
              </widget>
             </item>
             <item row="6" column="1">
              <widget class="QSpinBox" name="sb_number_ids_start_2">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="prefix">
                <string>IDS=</string>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="singleStep">
                <number>10</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
             <item row="13" column="0" colspan="2">
              <widget class="QPushButton" name="pb_convert_tabs_to_spaces_2">
               <property name="text">
                <string>Convert Tabs To Spaces</string>
               </property>
              </widget>
             </item>
             <item row="12" column="0">
              <widget class="QLabel" name="lb_tab_width_2">
               <property name="text">
                <string>Tab Width</string>
               </property>
              </widget>
             </item>
             <item row="12" column="1">
              <widget class="QSpinBox" name="sb_tab_width_2">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="maximum">
                <number>8</number>
               </property>
               <property name="singleStep">
                <number>2</number>
               </property>
               <property name="value">
                <number>4</number>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QSpinBox" name="sb_number_lines_step_2">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="singleStep">
                <number>10</number>
               </property>
               <property name="value">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="7" column="1">
              <widget class="QSpinBox" name="sb_number_ids_step_2">
               <property name="alignment">
                <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>1</number>
               </property>
              </widget>
             </item>
             <item row="3" column="0" colspan="2">
              <widget class="QPushButton" name="pb_number_lines_2">
               <property name="text">
                <string>Number Lines</string>
               </property>
              </widget>
             </item>
             <item row="15" column="0" colspan="2">
              <widget class="QPushButton" name="pb_remove_trailing_whitespace_2">
               <property name="text">
                <string>Remove Trailing Whitespace</string>
               </property>
              </widget>
             </item>
             <item row="14" column="0" colspan="2">
              <widget class="Line" name="ln_functions_6">
               <property name="frameShadow">
                <enum>QFrame::Plain</enum>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="2" column="1" colspan="2">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="0" rowspan="2">
           <widget class="QPlainTextEdit" name="pte_folder_preview">
            <property name="font">
             <font>
              <family>Cascadia Mono Light</family>
              <pointsize>10</pointsize>
             </font>
            </property>
            <property name="lineWrapMode">
             <enum>QPlainTextEdit::NoWrap</enum>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="tabStopDistance">
             <double>40.000000000000000</double>
            </property>
            <property name="placeholderText">
             <string>Folder Preview</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
